packages/nx/src/nx-init/add-nx-to-nest.ts

```diff
@@ -0,0 +1,339 @@
+
+  const isMonorepo = nestCLIJson['monorepo'] || false;
+  const webpackOptions = nestCLIJson['compilerOptions']?.['webpack']
+    ? {
+        enabled: true,
+        configPath: nestCLIJson['compilerOptions']['webpackConfigPath'],
+      }
+    : null;
+  const entryFile = nestCLIJson['entryFile'] || 'main';
+  const plugins = nestCLIJson['compilerOptions']?.['plugins'] || [];
+
+  return {
+    isJS: nestCLIJson['language'] === 'js',
+    isMonorepo,
+    webpackOptions,
+    entryFile,
+    plugins,
+    sourceRoot: nestCLIJson['sourceRoot'],
+  };
+}
+
+function addNestPluginToPackageJson(repoRoot: string) {
+  const path = joinPathFragments(repoRoot, `package.json`);
+  const json = readJsonFile(path);
+  json.devDependencies['@nrwl/nest'] = require('../../package.json').version;
+  writeJsonFile(path, json);
+}
+
+function createProjectJson(
+  repoRoot: string,
+  packageName: string,
+  nestCLIOptions: NestCLIOptions
+) {
+  const path = joinPathFragments(repoRoot, 'project.json');
+  const json = {
+    $schema: 'node_modules/nx/schemas/project-schema.json',
+    name: packageName,
+    sourceRoot: nestCLIOptions.sourceRoot,
+    projectType: 'application',
+    targets: {},
+    tags: [],
+  };
+
+  json.targets['serve'] = {
+    executor: '@nrwl/js:node',
+    options: {
+      buildTarget: `${packageName}:build`,
+    },
+  };
+
+  if (nestCLIOptions.webpackOptions) {
+    json.targets['build'] = {
+      executor: '@nrwl/webpack:webpack',
+      outputs: ['{options.outputPath}'],
+      options: {
+        target: 'node',
+        compiler: 'tsc',
+        outputPath: `dist/${packageName}`,
+        main: joinPathFragments(
+          nestCLIOptions.sourceRoot,
+          nestCLIOptions.entryFile
+        ),
+        tsConfig: 'tsconfig.build.json',
+      },
+      configurations: {
+        production: {
+          optimization: true,
+          extractLicenses: true,
+          inspect: false,
+        },
+      },
+    };
+    json.targets['serve'] = {
+      ...json.targets['serve'],
+      configurations: {
+        production: {
+          buildTarget: `${packageName}:build:production`,
+        },
+      },
+    };
+  } else {
+    json.targets['build'] = {
+      executor: '@nrwl/js:tsc',
+      outputs: ['{options.outputPath}'],
+      options: {
+        outputPath: `dist/${packageName}`,
+        main: joinPathFragments(
+          nestCLIOptions.sourceRoot,
+          nestCLIOptions.entryFile
+        ),
+        tsConfig: 'tsconfig.build.json',
+      },
+    };
+    json.targets['serve'] = {
+      ...json.targets['serve'],
+      configurations: {
+        debug: {
+          inspect: 'inspect',
+        },
+      },
+    };
+
+    // if we're using nrwl/js, then we add nrwl/js analyzeSourceFiles to nx.json
+    addNrwlJsPluginsConfig(repoRoot);
+  }
+
+  // lint
+  // TODO: nestjs cli uses parserOptions with __dirname -> .eslintrc.js is a requirement. What can we do?
```

Comment:
What's not working here? We won't be able to update `.eslintrc.js` as easily, but I'd expect that it works since we just call eslint under the hood.

Reply:
Everything is fine here. Our `nrwl/linter` generates a new base `.eslintrc` based on `.eslintrc.js` and use that as base for other ESLint configuration for libs.